aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/S3ClientOptions.java

```diff
@@ -103,4 +143,22 @@ public S3ClientOptions withPathStyleAccess(boolean pathStyleAccess) {
       return this;
     }
 
-}
\ No newline at end of file
+    /**
+     * <p>
+     * Configures the client to disbale chunked encoding for all requests.
+     * </p>
+     * <p>
+     * The default behaviour is to detect API request which enables chunked
+     * encoding automatically for PutObjectRequest and UploadPartRequest.
+     * Setting this flag will result in disabling chunked encoding for all
+     * requests.
+     * </p>
+     * @return The updated S3ClientOptions object with chunked encoding
+     *         disabled setting.
+     */
+    public S3ClientOptions disableChunkedEncoding() {
```

Comment:
It just seems like it would be surprising if a customer has explicitly disabled chunked encoding but it was magically re-enabled in the auth error retry strategy. You should be able to pass the client options in the constructor of this class. It's created in the invoke method of AmazonS3Client.